src/main/java/build/buildfarm/common/redis/RedisPriorityQueue.java

```diff
@@ -118,8 +145,15 @@ public boolean offer(String val) {
    */
   @Override
   public boolean offer(String val, double priority) {
-    jedis.zadd(name, priority, clock.millis() + ":" + val);
-    return true;
+    String stampedVal = clock.millis() + ":" + val;
+    if (maxNodeQueueDepth == UNLIMITED_QUEUE_DEPTH) {
+      jedis.zadd(name, priority, stampedVal);
+      return true;
+    }
+    List<String> keys = ImmutableList.of(name);
+    List<String> args =
+        ImmutableList.of(String.valueOf(maxNodeQueueDepth), String.valueOf(priority), stampedVal);
+    return String.valueOf(pushScript.eval(jedis, keys, args)) == "t";
```

Comment:
Switch this and the script return up from a "t" string to an integer return, indicating how many items were added (similar to https://redis.io/docs/latest/commands/zadd/#resp3-reply). Return 0 if it could not be offered, and interpret as such.